File: 1-spring-introductory/hello-spring/src/main/java/hello/hi/hellospring/repository/SpringDataJpaMemberRepository.java
package hello.hi.hellospring.repository;

import hello.hi.hellospring.domain.Member;
import org.springframework.data.jpa.repository.JpaRepository;
import java.util.Optional;

public interface SpringDataJpaMemberRepository extends JpaRepository<Member, Long>, MemberRepository {
    Optional<Member> findByName(String name);
}

File: 1-spring-introductory/hello-kopring/src/main/kotlin/com/example/hellokopring/repository/SpringDataJpaMemberRepository.kt
package com.example.hellokopring.repository

import com.example.hellokopring.domain.Member
import org.springframework.data.jpa.repository.JpaRepository

interface SpringDataJpaMemberRepository : JpaRepository<Member, Long>, MemberRepository {
    override fun findByName(name: String): Member?
}
